src/DepositManager.mo

```diff
@@ -52,95 +51,58 @@ module {
+        surcharge = feeManager.surcharge();
       };
       flow = {
-        credited = Int.abs(credit.flow);
+        credited = totalCredited;
       };
       totalConsolidated = totalConsolidated;
       funds = {
-        deposited = depositMap.sum() + underwayFunds;
+        deposited = map.depositSum() + underwayFunds;
         underway = underwayFunds;
-        queued = depositMap.sum();
+        queued = map.depositSum();
       };
-      nDeposits = depositMap.size();
-      nLocks = depositMap.locks();
+      nDeposits = map.depositsCount();
+      nLocks = map.locks();
     };
 
     /// Pause or unpause notifications.
     public func pause(b : Bool) = paused := b;
 
     /// Sets new surcharge amount.
-    public func setSurcharge(s : Nat) {
-      log(Principal.fromBlob(""), #surchargeUpdated({ old = surcharge(); new = s }));
-      depositRegistry.updateFee(icrc84.fee() + s, credit.changeUser);
-    };
+    // public func setSurcharge(s : Nat) {
+    //   log(Principal.fromBlob(""), #surchargeUpdated({ old = surcharge(); new = s }));
+    //   depositRegistry.updateFee(icrc84.fee() + s, credit.changeUser);
+    // };
 
     /// Retrieves the deposit of a principal.
-    public func getDeposit(p : Principal) : ?Nat = depositMap.getOpt(p);
-
-    /// Object to track all credit flows initiated from within DepositManager.
-    /// The flow should never become negative.
-    let credit = object {
-      public var flow : Int = 0;
-      public func change(who : { #user : Principal; #pool }, amount : Int) {
-        changeCredit(who, amount);
-        flow += amount;
-        if (flow < 0) trap("credit flow went negative");
-      };
-      public func changeUser(p : Principal, amount : Int) = change(#user p, amount);
-      public func changePool(amount : Int) = change(#pool, amount);
-    };
-
-    // get informed by updated ledger fee
-    public func registerFeeChange(oldFee : Nat, newFee : Nat) {
-      assert oldFee != newFee;
-      // set new deposit fee such that the surcharge remains the same
-      depositRegistry.updateFee(depositRegistry.fee + newFee - oldFee, credit.changeUser);
-      log(Principal.fromBlob(""), #feeUpdated({ old = oldFee; new = newFee }));
-    };
-
-    func process_deposit(p : Principal, deposit : Nat, release : ?Nat -> Int) : (Nat, Nat) {
-      if (deposit <= depositRegistry.fee) {
-        ignore release(null);
-        return (0, 0);
-      };
-      let delta = release(?deposit);
-      if (delta < 0) trap("latestDeposit < prevDeposit on notify");
-      if (delta == 0) return (0, 0);
-      let inc = Int.abs(delta);
-
-      let creditInc : Nat = switch (deposit == inc) {
-        case true { deposit - depositRegistry.fee }; // former value in DepositRegistry was 0
-        case false { inc }; // former value in DepositRegistry was > 0
-      };
-
-      credit.changeUser(p, creditInc);
-      (inc, creditInc);
+    public func getDeposit(p : Principal) : ?Nat = switch (map.getOpt(p)) {
```

Comment:
Do we still need this function? Because lib.mo has access to `data` it could do this directly.